File: leetcode/049_groupanagrams.py
class Solution(object):
    def groupAnagrams(self, strs):
        """
        :type strs: List[str]
        :rtype: List[List[str]]
        """
        d = dict()
        for s in strs:
            tmp = ''.join(sorted(s))
            if tmp in d:
                d[tmp].append(s)
            else:
                d[tmp] = [s]
        return d.values()

File: leetcode/049_groupanagrams.ts
function groupAnagrams(strs: string[]): string[][] {
    const dict: {[id: string]: string[]} = {}
    strs.forEach((s: string) => {
        const inOrder = s.split('').sort().join('')
        if (dict[inOrder]) {
            dict[inOrder].push(s)
        } else {
            dict[inOrder] = [s]
        }
    })
    return Object.values(dict)
};
